pkg/sse/sse.go

```diff
@@ -26,17 +27,25 @@ type Writer struct {
 func (w *Writer) Write(p []byte) (int, error) {
 	w.Lock()
 	defer w.Unlock()
-
-	if _, err := w.Writer.Write([]byte("data: ")); err != nil {
-		return 0, err
-	}
-	if _, err := w.Writer.Write(p); err != nil {
-		return 0, err
+	for _, line := range bytes.Split(p, []byte("\n")) {
+		data := fmt.Sprintf("data: %s\n", line)
+		if _, err := w.Writer.Write([]byte(data)); err != nil {
+			return 0, err
+		}
 	}
-	_, err := w.Writer.Write([]byte("\n\n"))
+	// add a terminating newline
+	_, err := w.Writer.Write([]byte("\n"))
```

Comment:
Yes, since the current message packet might be split into multiple lines, separation between events is done using a double newline. https://developer.mozilla.org/en-US/docs/Server-sent_events/Using_server-sent_events